subprojects/test-runner/service/src/main/kotlin/com/avito/runner/service/worker/device/Device.kt

```diff
@@ -35,9 +35,16 @@ interface Device {
 
     /**
      * @return `to` path
+     *
+     * todo deprecate in favor of [pullDirRecursively]
      */
     fun pull(from: Path, to: Path): Result<File>
 
+    /**
+     * @return `hostDir` path
+     */
+    fun pullDirRecursively(deviceDir: Path, hostDir: Path, validate: () -> Boolean): Result<File>
```

Comment:
Why `validate` doesn't operate file or dir?

Reply:
oh, it's just a draft to express that this validation should be outsourced because `AdbDevice` shouldn't know about pulled content

Reply:
Ok, will wait till ready to review